test/parallel/test-stream-readable-async-iterators.js

```diff
@@ -11,17 +11,6 @@ const {
 const assert = require('assert');
 
 async function tests() {
-  {
-    const AsyncIteratorPrototype = Object.getPrototypeOf(
-      Object.getPrototypeOf(async function* () {}).prototype);
-    const rs = new Readable({
-      read() {}
-    });
-    assert.strictEqual(
-      Object.getPrototypeOf(Object.getPrototypeOf(rs[Symbol.asyncIterator]())),
-      AsyncIteratorPrototype);
-  }
```

Comment:
Not sure why this fails... but it I don't think you can implement async iteration in a more "correct" way than async generator.